ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -86,44 +88,37 @@ class MySqlNamedBlobDb implements NamedBlobDb {
    * Select a record that matches a blob name (lookup by primary key).
    */
   private static final String GET_QUERY =
-      String.format("SELECT %s, %s, %s FROM %s WHERE %s", BLOB_ID, EXPIRES_TS, DELETED_TS, NAMED_BLOBS, PK_MATCH);
+      String.format("SELECT %s, %s, %s, %s FROM %s WHERE %s AND %s ORDER BY %s DESC LIMIT 1", BLOB_ID, VERSION,
+          EXPIRES_TS, DELETED_TS, NAMED_BLOBS, PK_MATCH, STATE_MATCH, VERSION);
 
   /**
    * Select records up to a specific limit where the blob name starts with a string prefix. The fourth parameter can
    * be used for pagination.
    */
   private static final String LIST_QUERY = String.format("SELECT %1$s, %2$s, %3$s, %4$s FROM %5$s "
-          + "WHERE (%6$s, %7$s) = (?, ?) AND %1$s LIKE ? AND %1$s >= ? ORDER BY %1$s ASC LIMIT ?", BLOB_NAME, BLOB_ID,
-      EXPIRES_TS, DELETED_TS, NAMED_BLOBS, ACCOUNT_ID, CONTAINER_ID);
+          + "WHERE (%6$s, %7$s) = (?, ?) AND %8$s AND %1$s LIKE ? AND %1$s >= ? ORDER BY %1$s ASC LIMIT ?", BLOB_NAME,
+      BLOB_ID, EXPIRES_TS, DELETED_TS, NAMED_BLOBS, ACCOUNT_ID, CONTAINER_ID, STATE_MATCH);
 
   /**
    * Attempt to insert a new mapping into the database.
    */
   private static final String INSERT_QUERY =
-      String.format("INSERT INTO %s (%s, %s, %4$s, %5$s, %6$s) VALUES (?, ?, ?, ?, ?)", NAMED_BLOBS, ACCOUNT_ID,
-          CONTAINER_ID, BLOB_NAME, BLOB_ID, EXPIRES_TS);
-
-  /**
-   * If a record already exists for a named blob, attempt an update if the record in the DB represents an expired or
-   * deleted blob.
-   */
-  private static final String UPDATE_IF_DELETED_OR_EXPIRED_QUERY =
```

Comment:
why did u remove UPDATE_IF_DELETED_OR_EXPIRED_QUERY

Reply:
Because it's not needed anymore, previously we assume there is only one row per blob_id, so this 'UPDATE_IF_DELETED_OR_EXPIRED_QUERY' was applied when the row for a blob_id 'AlreadyExist', but now we are using blob_id+version, that logic is not needed anymore.